cmd/demoserver/main_test.go

```diff
@@ -66,43 +66,37 @@ func TestElizaServer(t *testing.T) {
 		for _, client := range clients {
 			sendValues := []string{"Hello!", "How are you doing?", "I have an issue with my bike", "bye"}
 			var receivedValues []string
-			stream := client.Converse(context.Background())
-			var wg sync.WaitGroup
-			wg.Add(2)
-			errs := make(chan error, 4)
-			go func() {
-				defer wg.Done()
+			grp, ctx := errgroup.WithContext(context.Background())
+			stream := client.Converse(ctx)
+			grp.Go(func() error {
 				for _, sentence := range sendValues {
 					err := stream.Send(&elizav1.ConverseRequest{Sentence: sentence})
-					errs <- err
+					if err != nil {
+						return err
+					}
 				}
 				err := stream.CloseRequest()
-				errs <- err
-			}()
-			go func() {
-				defer wg.Done()
+				if err != nil {
+					return err
+				}
+				return nil
```

Comment:
This can be simplified to just `return stream.CloseRequest()` (in the goroutine below as well).